pkg/controller/framework/controller.go

```diff
@@ -102,3 +103,117 @@ func (c *Controller) processLoop() {
 		}
 	}
 }
+
+// ResourceEventHandler can handle notifications for events that happen to a resource.
+// The events are informational only, so you can't return an error.
+// OnDelete will get the final state of the item if it is known, otherwise it will get
```

Comment:
If you're going to describe the semantics of OnDelete (which I think is good), then I would suggest also doing the same for OnAdd() and OnUpdate(). BTW, is there some way OnUpdate() could get a copy of both the old and new value, instead of just the new value?